### DISCUSSION TITLE
Developer Control Pane - Source/License Clarification

### DISCUSSION THREAD

**Discussion by swimmesberger** [microsoft/aspire#1913] [General]
Aspire uses the developer control pane (DCP) underneath to control local development/testing orchestration (Aspire.Hosting.Orchestration;github.com/microsoft/usvc-apiserver). 
Are there any plans to open-source that tool too? Currently the EULA states "You may install and use any number of copies of the software to develop and test your applications", is it allowed to use DCP in on-premise, "local-cloud", single-node scenarios too? This makes it possible to generally develop distributed applications in a unified way independent of the production requirement regarding reliability. I know there are tools like minikube, k3s for those scenarios too but dcp might be easier and more comfortable to use.

**Comment by d0pare** [comment]
It seems like they are planning to make it a commercial product.

> These license terms are effective by you using the software and terminates following first availability of a commercial release of the software or upon termination by Microsoft. Microsoft may extend this agreement in its discretion.  

It also collects data and there is no documentation on how you can opt out of it.

**Comment by danmoseley** [comment]
> It also collects data and there is no documentation on how you can opt out of it.

@d0pare what are you referring to here?

**Comment by d0pare** [comment]
It's from MICROSOFT DEVELOPER CONTROL PLANE  EULA
> 6.	DATA COLLECTION. The software may collect information about you and your use of the software, and send that to Microsoft. Microsoft may use this information to provide services and improve our products and services. You may opt-out of many of these scenarios, but not all, as described in the software documentation. There are also some features in the software that may enable you and Microsoft to collect data from users of your applications. If you use these features, you must comply with applicable law, including providing appropriate notices to users of your applications together with a copy of Microsoft’s privacy statement. Our privacy statement is located at https://aka.ms/privacy. You can learn more about data collection and its use from the software documentation and our privacy statement. Your use of the software operates as your consent to these practices. 

You can check EULA.rtf in the following Nuget package under the tools folder.
https://nuget.info/packages/Aspire.Hosting.Orchestration.win-x64/8.0.0-preview.1.23557.2

**Comment by spboyer** [comment]
@d0pare DCP does not currently collect any data or telemetry, data or usage.

**Comment by mayuki** [comment]
Officially, Aspire is described as [“community-driven”](https://devblogs.microsoft.com/aspire/aspirepolyglot/#:~:text=become%20a%20thriving%2C-,community%2Ddriven,-%2C%20extensible%2C%20code%2Dfirst) and [“Free and open source”](https://aspire.dev/#:~:text=Free%20and%20open%20source), but the Developer Control Plane (dcp.exe) which is a key component for local development (AppHost), currently appears to be proprietary.

Does this mean DCP is not part of Aspire? Or is there a plan to release its source code in the future?

**Reply by jaredthirsk** [reply]
> Officially, Aspire is described as [“community-driven”](https://devblogs.microsoft.com/aspire/aspirepolyglot/#:~:text=become%20a%20thriving%2C-,community%2Ddriven,-%2C%20extensible%2C%20code%2Dfirst) and [“Free and open source”](https://aspire.dev/#:~:text=Free%20and%20open%20source), but the Developer Control Plane (dcp.exe) which is a key component for local development (AppHost), currently appears to be proprietary.
> 
> Does this mean DCP is not part of Aspire? Or is there a plan to release its source code in the future?

"DCP is at the core of the Aspire AppHost orchestration functionality." (https://aspire.dev/architecture/overview/#developer-control-plane)

So for the core of Aspire AppHost orchestration functionality: "You may not: ... attempt to derive the source code for the software."  (from EULA.rtf in nuget package: https://www.nuget.org/packages/Aspire.Hosting.Orchestration.win-x64/13.1.0.)

Somebody please make it make sense.

**Reply by teo-tsirpanis** [reply]
Furthermore, the `EULA.rtf` in the package contradicts its declared `MIT` license expression. `EULA.rtf` should be rewritten in plain text and set as the packages' license file, so that it's visible from the NuGet gallery and tooling.

<img width="1174" height="557" alt="image" src="https://github.com/user-attachments/assets/6d4951f8-b5a4-4c8d-9789-6e76b410807e" />

**Comment by joshmakestuff** [comment]
Was hoping someone could clarify this.

**Comment by edandersen** [comment]
I made a video about this issue if anyone is interested! https://www.youtube.com/watch?v=cZntPxnnmSI

**Comment by danielwcarey** [comment]
I hope this gets addressed soon. I cannot in good faith recommend this in the industry I work in. I personally have some reservations now, but I also really have strong, positive opinions about using this technology and want to use Aspire for **every new project**. I have an extreme measure of respect for this development team and understand that "It's complicated", but this revelation has created some great concerns. Chain of custody[and Software Bill of Materials] is important to me both personally and professionally and again, hope this can be worked out in the near term.

I'd prefer the solution be an open-source C# replacement or secondly, open-source the Go code.

**Comment by davidfowl** [comment]
It's open and MIT https://github.com/microsoft/dcp

**Reply by danielwcarey** [reply]
Very grateful for you all. Thank You!
